production/install

```diff
@@ -551,119 +562,169 @@ osCertbotDryRun()
         done
     fi
 
     # electrs testnet data
     if [ "${BITCOIN_TESTNET_ENABLE}" = ON ];then
+        zfsCleanupDataset "${ELECTRS_TESTNET_ZPOOL}/electrs/testnet"
         zfs create -o "mountpoint=${ELECTRS_TESTNET_DATA}" "${ELECTRS_TESTNET_ZPOOL}/electrs/testnet"
         for folder in cache history txstore
         do
+            zfsCleanupDataset "${ELECTRS_TESTNET_ZPOOL}/electrs/testnet/${folder}"
             zfs create -o "mountpoint=${ELECTRS_TESTNET_DATA}/newindex/${folder}" "${ELECTRS_TESTNET_ZPOOL}/electrs/testnet/${folder}"
         done
     fi
 
     # electrs testnet4 data
     if [ "${BITCOIN_TESTNET4_ENABLE}" = ON ];then
+        zfsCleanupDataset "${ELECTRS_TESTNET4_ZPOOL}/electrs/testnet4"
         zfs create -o "mountpoint=${ELECTRS_TESTNET4_DATA}" "${ELECTRS_TESTNET4_ZPOOL}/electrs/testnet4"
         for folder in cache history txstore
         do
+            zfsCleanupDataset "${ELECTRS_TESTNET4_ZPOOL}/electrs/testnet4/${folder}"
             zfs create -o "mountpoint=${ELECTRS_TESTNET4_DATA}/newindex/${folder}" "${ELECTRS_TESTNET4_ZPOOL}/electrs/testnet4/${folder}"
         done
     fi
 
     # electrs signet data
     if [ "${BITCOIN_SIGNET_ENABLE}" = ON ];then
+        zfsCleanupDataset "${ELECTRS_SIGNET_ZPOOL}/electrs/signet"
         zfs create -o "mountpoint=${ELECTRS_SIGNET_DATA}" "${ELECTRS_SIGNET_ZPOOL}/electrs/signet"
         for folder in cache history txstore
         do
+            zfsCleanupDataset "${ELECTRS_SIGNET_ZPOOL}/electrs/signet/${folder}"
             zfs create -o "mountpoint=${ELECTRS_SIGNET_DATA}/newindex/${folder}" "${ELECTRS_SIGNET_ZPOOL}/electrs/signet/${folder}"
         done
     fi
 
     # electrs liquid data
     if [ "${ELEMENTS_LIQUID_ENABLE}" = ON ];then
+        zfsCleanupDataset "${ELECTRS_LIQUID_ZPOOL}/electrs/liquid"
         zfs create -o "mountpoint=${ELECTRS_LIQUID_DATA}" "${ELECTRS_LIQUID_ZPOOL}/electrs/liquid"
         for folder in cache history txstore
         do
+            zfsCleanupDataset "${ELECTRS_LIQUID_ZPOOL}/electrs/liquid/${folder}"
             zfs create -o "mountpoint=${ELECTRS_LIQUID_DATA}/newindex/${folder}" "${ELECTRS_LIQUID_ZPOOL}/electrs/liquid/${folder}"
         done
     fi
 
     # electrs liquidtestnet data
     if [ "${ELEMENTS_LIQUIDTESTNET_ENABLE}" = ON ];then
+        zfsCleanupDataset "${ELECTRS_LIQUIDTESTNET_ZPOOL}/electrs/liquidtestnet"
         zfs create -o "mountpoint=${ELECTRS_LIQUIDTESTNET_DATA}" "${ELECTRS_LIQUIDTESTNET_ZPOOL}/electrs/liquidtestnet"
         for folder in cache history txstore
         do
+            zfsCleanupDataset "${ELECTRS_LIQUIDTESTNET_ZPOOL}/electrs/liquidtestnet/${folder}"
             zfs create -o "mountpoint=${ELECTRS_LIQUIDTESTNET_DATA}/newindex/${folder}" "${ELECTRS_LIQUIDTESTNET_ZPOOL}/electrs/liquidtestnet/${folder}"
         done
     fi
 
     if [ "${CLN_INSTALL}" = ON ];then
+        zfsCleanupDataset "${ZPOOL}/cln"
```

Comment:
No, you cannot just yolo nuke this filesystem, we need to preserve the data and continue with an upgrade. I believe @portlandhodl already worked on this, so maybe you can check his WIP work branch and base on top of that